app/services/confirm_deferred_offer.rb

```diff
@@ -0,0 +1,22 @@
+class ConfirmDeferredOffer
```

Comment:
More of a question for myself – why does this service not include an auth method like the others?

Reply:
This is a wrapper around two existing services who already handle authentication internally. I could remove duplication and move the authentication checks here, but having all checks (authentication + other validation errors) take place at the same level makes it easier to figure out the possible outcome of calling ReinstateConditionsMet or ReinstatePendingConditions.